File: 2390-removing-stars-from-a-string/2390-removing-stars-from-a-string.py
class Solution:
    def removeStars(self, s: str) -> str:
        i = 0  
        s = list(s)  
        for c in s:  
            if c != '*':  
                s[i] = c
                i += 1  
            else:  
                i -= 1
        return ''.join(s[:i])         

File: 2390-removing-stars-from-a-string/2390-removing-stars-from-a-string.rs
impl Solution {
    pub fn remove_stars(s: String) -> String {
        let mut res = String::new();
        let mut number_of_stars = 0;

        for c in s.chars().rev() {
            if c == '*' {
                number_of_stars += 1;
            } else if number_of_stars > 0 {
                number_of_stars -= 1;
            } else {
                res.push(c);
            }
        }

        res.chars().rev().collect()
    }
}